### DISCUSSION TITLE
Seeking help with saral backend layout configuration

### DISCUSSION THREAD

**Discussion by ManojNathIC** [Sunbird-Saral/Community#71] [Q&A]
Hello everyone,

I'm currently working on a setup Saral backend and need assistance on layout configuration,

What is the request payload to create a layout config or Layout Generation Steps?

I would appreciate it if anyone could provide some sample layout files and what is the curl command to upload content for those layout files.

Steps followed so far:

* Installed backend and explored swagger documentation:
 
> Able to create ROI and get that created ROI based on examId and set.

* Which tool you are using to tag ROIs with the layout? One which is mentioned on docs showing archived by the owner, Please suggest a tool for the same.

Thank you in advance for your time and assistance.

**Comment by ShreyashiSoni** [comment]
Hi @dileep-gadiraju ! Can you please assist Manoj on the above queries. Thanks!

**Comment by dileep-gadiraju** [comment]
Hi @ManojNathIC, please refer to https://saral.sunbird.org/use/layout-configuration for the overall process. 
You can use https://github.com/microsoft/VoTT tool for tagging the layout.

**Reply by ManojNathIC** [reply]
Thank you @dileep-gadiraju
Now I'm able to tag the layout.
Whenever we transform raw VoTT Json to target layout configuration JSON format, do we need to customize the jupyter notebook file based on the number of questions?

**Reply by dileep-gadiraju** [reply]
Hi @ManojNathIC , 

Good to know that you are able to tag your layout.  As you have gone through creating layouts with supported extraction methods/validations,  a variety of layouts can be designed. For example, refer to a few of the below layouts certified, they are very different from each other. So Jupyter Notebook shared, you can re-use core functions but as each layout can be very different, you may need to customize the script also accordingly depending on the requirement. Hope this clarifies, please feel free to ask if you have any follow-up queries.

https://github.com/Sunbird-Saral/Project-Saral/blob/v1-develop/docs/layout-design/Design/BENEFICIARY%20%26%20Growth%20Layout.pdf

https://github.com/Sunbird-Saral/Project-Saral/tree/v1-develop/docs/layout-design/Design#:~:text=School%20Evaluation.pdf

**Comment by kk9** [comment]
> Hello everyone,
> 
> I'm currently working on a setup Saral backend and need assistance on layout configuration,
> 
> What is the request payload to create a layout config or Layout Generation Steps?
> 
> I would appreciate it if anyone could provide some sample layout files and what is the curl command to upload content for those layout files.
> 
> Steps followed so far:
> 
> * Installed backend and explored swagger documentation:
> 
> > Able to create ROI and get that created ROI based on examId and set.
> 
> * Which tool you are using to tag ROIs with the layout? One which is mentioned on docs showing archived by the owner, Please suggest a tool for the same.
> 
> Thank you in advance for your time and assistance.

Hello Everyone, 

I have attached a layout that we use, can you please guide me on the following, this is related to same project so I assuming this is an extended question, let me know if you want me to take this in new tread.

Can use this layout Saral without any modifications? If not, what are the modifications needed in saral side?
[3. Readiness Mela_Report Card - Hindi7.2.23.pdf](https://github.com/Sunbird-Saral/Community/files/11199768/3.Readiness.Mela_Report.Card.-.Hindi7.2.23.pdf)

**Reply by dileep-gadiraju** [reply]
Hi @kk9 ,

Please refer to https://saral.sunbird.org/use/layout-design-guidelines to design a saral compatible and printable layout.  
Once you design a saral compatible and printable layout, you can generate ROI with the above provided steps and configure it in the backend. No change needs to be done in Saral App or SDK.

**Comment by kk9** [comment]
great and thanks, will check link and  do the changes accordingly.

**Comment by dileep-gadiraju** [comment]
Hi @kk9 / @ManojNathIC ,

Please check and mark this thread answered if all your queries are addressed. Thanks.
